Framework/Core/include/Framework/DataOutputDirector.h

```diff
@@ -106,14 +108,25 @@ struct DataOutputDescriptor {
   }
 
   void setFilename(std::string fn) { filename = fn; }
+  void setFilename(std::string* fnptr) { dfnptr = fnptr; }
+  std::string getFilename()
+  {
+    if (filename.empty() && dfnptr)
+      return (std::string)*dfnptr;
+    else
+      return filename;
+  }
 
   void printOut()
   {
     LOG(INFO) << "DataOutputDescriptor";
     LOG(INFO) << "  table name: " << tablename.c_str();
-    LOG(INFO) << "  file name : " << filename.c_str();
+    LOG(INFO) << "  file name : " << getFilename().c_str();
     LOG(INFO) << "  tree name : " << treename.c_str();
-    LOG(INFO) << "  columns   : " << colnames.size();
+    if (colnames.size() == 0)
+      LOG(INFO) << "  columns   : all";
+    else
+      LOG(INFO) << "  columns   : " << colnames.size();
```

Comment:
Something like:

```
LOG(INFO) << "  columns   : " << ? colnames.empty() ? "all" : std::to_string(colnames.size());
```

also you can use the `LOGP(INFO, pattern, substitutions...)`. E.g.:

LOGP(INFO, "DataOutputDescriptor : table name {} : file name {}", table, file);

Reply:
done